datumaro/plugins/icdar_format/converter.py

```diff
@@ -94,15 +95,11 @@ def _save_item(self, subset_name, subset, item):
                         text = ann.attributes['text']
                     if text == ' ':
                         annotation += '#'
-                    if 'color' in ann.attributes and \
-                            len(ann.attributes['color'].split()) == 3:
-                        color = ann.attributes['color'].split()
-                        colormap.append(
-                            (int(color[0]), int(color[1]), int(color[2])))
-                        annotation += ' '.join(p for p in color)
-                    else:
-                        raise Exception("Item %s: a mask must have "
-                            "an RGB color attribute, e.g. '10 7 50'" % item.id)
+                    color = ann.attributes.get('color', '').split()
+                    color = tuple(map(int, color)) if len(color) == 3 \
+                        else gen_colormap.pop(i)
+                    colormap.append(color)
```

Comment:
The generated colors may overlap with the custom ones. Probably, we need to avoid using the same colors for objects, unless explicitly requested. Check solution [here](https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/plugins/transforms.py#L656-L669) for the same problem.

Reply:
Done